python.d/python_modules/base.py

```diff
@@ -359,68 +417,57 @@ def check(self):
         else:
             return False
 
-    def create(self):
-        """
-        Create charts
-        :return: boolean
-        """
-        for name in self.order:
-            if name not in self.charts:
-                continue
-            self.definitions[name] = []
-            for line in self.charts[name]['lines']:
-                self.definitions[name].append(line['name'])
 
-        idx = 0
-        data = self._formatted_data()
-        if data is None:
-            return False
-        data_stream = ""
-        for name in self.order:
-            header = "CHART " + \
-                     self.__module__ + "_" + \
-                     self.name + "." + \
-                     name + " " + \
-                     self.charts[name]['options'] + " " + \
-                     str(self.priority + idx) + " " + \
-                     str(self.update_every)
-            content = ""
-            # check if server has this datapoint
-            for line in self.charts[name]['lines']:
-                if line['name'] in data:
-                    content += "\nDIMENSION " + line['name'] + " " + line['options']
+class LogService(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        # definitions are created dynamically in create() method based on 'charts' dictionary. format:
+        # definitions = {
+        #     'chart_name_in_netdata' : [ charts['chart_name_in_netdata']['lines']['name'] ]
+        # }
+        self.log_path = ""
+        self._last_position = 0
+        # self._log_reader = None
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        # FIXME Remove preventing of frequent log parsing
+        if self.timetable['freq'] < 3:
+            self.timetable['freq'] = 3
+        self.retries = 100000  # basically always retry
 
-            if len(content) > 0:
-                data_stream += header + content + "\n"
-                idx += 1
+    def _get_data(self):
+        lines = []
+        try:
+            if os.path.getsize(self.log_path) < self._last_position:
+                self._last_position = 0
+            with open(self.log_path, "r") as fp:
+                fp.seek(self._last_position)
+                for i, line in enumerate(fp):
+                    lines.append(line)
+                self._last_position = fp.tell()
```

Comment:
I have run time at 0 ms, so it shouldn't be very expensive. However I will add two lines to prevent opening file if it content hasn't changed.

Good to know about inner workings of `percentage-of-absolute-row`, will change it in next PR.